empty build args causes builds to fail

This ticket is a continuation of #1528, which I cannot confirm to be fixed.

When a build arg in the pom.xml is set to an empty string, docker-maven-plugin throws an error:
> [ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.45.1:build (default-cli) on project test: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.45.1:build failed: null value in entry: TESTARG=null -> [Help 1]

I do not see a way to legitimately set an arg to an empty string.

Reproduction pom.xml:
```xml
<project>
  <modelVersion>4.0.0</modelVersion>
  <groupId>testgroup</groupId>
  <artifactId>test</artifactId>
  <name>Test</name>
  <version>1.0.0-SNAPSHOT</version>
  <properties>
    <testproperty />
  </properties>
  <build>
    <plugins>
      <plugin>
        <groupId>io.fabric8</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <version>0.45.1</version>
        <configuration>
          <images>
            <image>
              <name>example-image</name>
              <build>
                <contextDir>${project.basedir}</contextDir>
                <args>
                  <TESTARG>${testproperty}</TESTARG>
                </args>
              </build>
            </image>
          </images>
        </configuration>
      </plugin>
    </plugins>
  </build>
</project>
```

Dockerfile:
```Dockerfile
FROM hello-world
```

(note: the `<contextDir>` and Dockerfile are completely irrelevant to the bug, it occurs no matter the content, even if `<contextDir>` is completely left out, or if the Dockerfile uses the ARG in any way or sets any default value)

failing Command (you might need to scroll right):
```bash
$ mvn docker:build
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------------< testgroup:test >---------------------------
[INFO] Building Test 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- docker:0.45.1:build (default-cli) @ test ---
[INFO] Building tar: C:\Users\jbrinkmann\work\Hilfsmittel\tmp-mvn-docker-bug\target\docker\example-image\tmp\docker-build.tar
[INFO] DOCKER> [example-image:latest]: Created docker-build.tar in 124 milliseconds
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.282 s
[INFO] Finished at: 2025-03-12T17:05:41+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.45.1:build (default-cli) on project test: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.45.1:build failed: null value in entry: TESTARG=null -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:       
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

It does work if the build arg is a non-empty string, e.g.:
```bash
$ mvn docker:build -Dtestproperty=1                                       
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------------< testgroup:test >---------------------------
[INFO] Building Test 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- docker:0.45.1:build (default-cli) @ test ---
[INFO] Building tar: C:\Users\jbrinkmann\work\Hilfsmittel\tmp-mvn-docker-bug\target\docker\example-image\tmp\docker-build.tar
[INFO] DOCKER> [example-image:latest]: Created docker-build.tar in 121 milliseconds
[INFO] DOCKER> [example-image:latest]: Built image sha256:bfbb0cc14f13f9ed1ae86abc2b9f11181dc50d779807ed3a3c5e55a6936dbdd5
[INFO] DOCKER> example-image: Removed dangling image sha256:498a000f370d8c37927118ed80afe8adc38d1edcbfc071627d17b25c88efcab0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.827 s
[INFO] Finished at: 2025-03-12T17:07:28+01:00
[INFO] ------------------------------------------------------------------------
```

I was able to reproduce this issue with several versions, such as 0.30.0, 0.39.0, 0.42.0, 0.43.3, 0.44.0, 0.45.1
```bash
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.39.0:build (default-cli) on project test: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.39.0:build failed: null value in entry: TESTARG=null ->
```

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -1,5 +1,6 @@
 * **0.49-SNAPSHOT**:
   - Add opt-in `<buildAllPlatforms>` buildx option to build all platforms during docker:build, warming the builder cache so a later docker:push reuses it ([#1866](https://github.com/fabric8io/docker-maven-plugin/issues/1866))
+  - Normalize empty `<args>` build argument values (e.g. from a property that resolves to an empty value) to an empty string instead of failing the build ([#1858](https://github.com/fabric8io/docker-maven-plugin/issues/1858))
 
 * **0.48.1 (2026-02-07)**:
   - Use wait config if no Docker Compose healthcheck ([1771](https://github.com/fabric8io/docker-maven-plugin/issues/1771))

--- a/src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java
+++ b/src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java
@@ -7,6 +7,7 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashSet;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
@@ -793,6 +794,8 @@ public class BuildImageConfiguration implements Serializable {
             healthCheck.validate();
         }
 
+        normalizeBuildArgs();
+
         ensureUniqueAssemblyNames(log);
 
         if (command != null) {
@@ -836,6 +839,21 @@ public class BuildImageConfiguration implements Serializable {
                 throw new IllegalArgumentException("Assembly names must be unique");
             }
         }
+    }
+
+    /**
+     * Build args coming from a Maven property that resolves to an empty or undefined value are injected
+     * as {@code null} (e.g. {@code <FOO>${someEmptyProperty}</FOO>}). Docker treats a build arg without a
+     * value as an empty string, so normalize {@code null} values to {@code ""} instead of letting the build
+     * fail later when the args are collected into a map that rejects null values. See #1858.
+     */
+    private void normalizeBuildArgs() {
+        if (args == null || !args.containsValue(null)) {
+            return;
+        }
+        Map<String, String> normalizedArgs = new LinkedHashMap<>();
+        args.forEach((key, value) -> normalizedArgs.put(key, value == null ? "" : value));
+        args = normalizedArgs;
     }
 
     // Initialize the dockerfile location and the build mode

--- a/src/test/java/io/fabric8/maven/docker/config/BuildImageConfigurationTest.java
+++ b/src/test/java/io/fabric8/maven/docker/config/BuildImageConfigurationTest.java
@@ -28,6 +28,8 @@ import org.mockito.junit.jupiter.MockitoExtension;
 import java.io.File;
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
 
 import static io.fabric8.maven.docker.config.ArchiveCompression.gzip;
 import static io.fabric8.maven.docker.config.ArchiveCompression.none;
@@ -48,6 +50,22 @@ class BuildImageConfigurationTest {
         BuildImageConfiguration config = new BuildImageConfiguration();
         config.initAndValidate(logger);
         Assertions.assertFalse(config.isDockerFileMode());
+    }
+
+    @Test
+    // Tests fix for #1858: a build arg whose value resolves to an empty/undefined property is
+    // injected as null and must be normalized to an empty string instead of failing the build
+    void buildArgWithNullValueIsNormalizedToEmptyString() {
+        Map<String, String> args = new HashMap<>();
+        args.put("EMPTY_ARG", null);
+        args.put("REAL_ARG", "value");
+        BuildImageConfiguration config =
+            new BuildImageConfiguration.Builder().args(args).build();
+
+        config.initAndValidate(logger);
+
+        Assertions.assertEquals("", config.getArgs().get("EMPTY_ARG"));
+        Assertions.assertEquals("value", config.getArgs().get("REAL_ARG"));
     }
 
     @Test

